index.ts

```diff
@@ -213,6 +214,13 @@ addTests('isDiscussion', [
 	'https://github.com/orgs/community/discussions/11202',
 ]);
 
+export const isNewDiscussion = (url: URL | HTMLAnchorElement | Location = location): boolean => (getRepo(url)?.path === 'discussions/new' || getOrg(url)?.path === 'discussions/new')
+	&& new URLSearchParams(url.search).get('category') !== null;
```

Comment:
Why? If you remove the category, it's redirected to https://github.com/withastro/roadmap/discussions/new/choose

Also https://github.com/sindresorhus/meta/discussions/7. I'm not sure why this lint rule isn't complaining https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-null.md

Reply:
> If you remove the category, it's redirected to [withastro/roadmap/discussions/new/choose](https://github.com/withastro/roadmap/discussions/new/choose)

Yes, But is [withastro/roadmap/discussions/new/choose](https://github.com/withastro/roadmap/discussions/new/choose) a new discussions? Because I'm referring to `isNewIssue()`, which does not match `issue/new/choose`, see [Test CI](https://github.com/refined-github/github-url-detection/actions/runs/5555969720/job/15050534278#step:5:5202) for more details.

> Also [sindresorhus/meta#7](https://github.com/sindresorhus/meta/discussions/7). I'm not sure why this lint rule isn't complaining [sindresorhus/eslint-plugin-unicorn@`main`/docs/rules/no-null.md](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-null.md?rgh-link-date=2023-07-23T12%3A55%3A08Z)

The `no-null` has an option: [checkStrictEquality](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/1b6757e80cdb7a120d90f2df5fef087a8b0f1cc0/docs/rules/no-null.md#checkstrictequality), which is set to [false](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/1b6757e80cdb7a120d90f2df5fef087a8b0f1cc0/docs/rules/no-null.md#L43) by default. It must be manually enabled:

```diff
  	"xo": {
		"overrides": [
			"rules": {
			...
+			"unicorn/no-null" : ["error", { "checkStrictEquality": false }],
			}
		]
	}
```

Reply:
> is [withastro/roadmap/discussions/new/choose](https://github.com/withastro/roadmap/discussions/new/choose) a new discussions

It's not and it's not matched by `getRepo(url)?.path === 'discussions/new' || getOrg(url)?.path === 'discussions/new'` already. Redirects are not taken into considerations because `github-url-detection` is meant to be used on real/final URLs.